dotnet-peer/Libp2pChat/Infrastructure/Logging/TerminalLogger.cs

```diff
@@ -0,0 +1,100 @@
+using Libp2pChat.Application.Interfaces;
+using Libp2pChat.Presentation.UI;
+using Microsoft.Extensions.Logging;
+
+namespace Libp2pChat.Infrastructure.Logging;
+
+/// <summary>
+/// A logger implementation that logs to the terminal UI.
+/// </summary>
+public class TerminalLogger : IAppLogger
+{
+    private readonly IChatUI _chatUI;
+    private readonly ILogger _logger;
+    private readonly string _category;
+    
+    /// <summary>
+    /// Creates a new instance of the <see cref="TerminalLogger"/> class.
+    /// </summary>
+    /// <param name="chatUI">The chat UI.</param>
+    /// <param name="logger">The underlying logger.</param>
+    /// <param name="category">The category name for the logger.</param>
+    public TerminalLogger(IChatUI chatUI, ILogger logger, string category)
+    {
+        _chatUI = chatUI;
+        _logger = logger;
+        _category = category;
+    }
+    
+    /// <inheritdoc />
+    public void LogInformation(string message)
+    {
+        string formattedMessage = $"[{DateTime.Now:HH:mm:ss.fff}] [Info] {message}";
+        _chatUI.AddLog(formattedMessage);
+        // _logger.LogInformation(message);
+    }
+    
+    /// <inheritdoc />
+    public void LogWarning(string message)
+    {
+        string formattedMessage = $"[{DateTime.Now:HH:mm:ss.fff}] [Warning] {message}";
```

Comment:
The body can be a separate function that logs using the same formatting, to reduce duplication